spec/services/shopify_refunder_spec.rb

```diff
@@ -1,53 +1,108 @@
 
-  context '.initialize' do
-    let(:transaction_amount) { 1 }
-    let(:credited_money) { 1 }
-
-    before do
-      allow(ShopifyAPI::Transaction).to receive(:find).and_return(transaction)
-    end
+  before do
+    allow(ShopifyAPI::Transaction).to receive(:find).and_return(transaction)
+  end
 
-    subject { described_class.new(credited_money, transaction_id, options) }
+  context '.initialize' do
+    subject { described_class.new(credited_money_in_cents, transaction_id, options) }
 
     it 'successfully does it\'s thing' do
       expect(subject).to be_truthy
     end
   end
 
   context '.perform' do
+    let(:pos_refund) { double('refund', errors: []) }
+
+    before do
+      allow(ShopifyAPI::Refund).to receive(:create).and_return(pos_refund)
+    end
+
     context 'when the shopify transaction is not found' do
+      before do
+        allow(ShopifyAPI::Transaction).to receive(:find).and_return(nil)
+      end
+
+      subject { described_class.new(credited_money_in_cents, transaction_id, options) }
+
       it 'throws an error' do
+        cause = ->{ subject.perform }
+        expect(&cause).to raise_error(ActiveMerchant::Billing::ShopifyGateway::TransactionNotFoundError)
       end
     end
 
     context 'with a full refund' do
-      let(:transaction_amount) { 1 }
-      let(:credited_money) { 1 }
-
-      subject { described_class.new(credited_money, transaction_id, options) }
+      subject { described_class.new(credited_money_in_cents, transaction_id, options) }
 
       it 'performs a refund on shopify' do
+        expect(ShopifyAPI::Refund).to receive(:create)
+        subject.perform
+      end
+
+      it 'returns an ActiveMerchant response' do
+        result = subject.perform
+        expect(result).to be_a(ActiveMerchant::Billing::Response)
+      end
+
+      it 'returns an ActiveMerchant succesfull response' do
+        result = subject.perform
+        expect(result.success?).to be_truthy
       end
     end
 
     context 'with a partial refund' do
+      let(:transaction_amount) { 2 }
+      let(:credited_money_in_cents) { 100 }
+
+      subject { described_class.new(credited_money_in_cents, transaction_id, options) }
+
       it 'performs a refund on shopify' do
+        expect(ShopifyAPI::Refund).to receive(:create)
+        subject.perform
+      end
+
+      it 'returns an ActiveMerchant response' do
+        result = subject.perform
+        expect(result).to be_a(ActiveMerchant::Billing::Response)
+      end
+
+      it 'returns an ActiveMerchant succesfull response' do
+        result = subject.perform
+        expect(result.success?).to be_truthy
       end
     end
 
     context 'with a credited amount bigger than the transaction' do
+      let(:transaction_amount) { 1 }
+      let(:credited_money_in_cents) { 200 }
+
+      subject { described_class.new(credited_money_in_cents, transaction_id, options) }
+
       it 'throws an error' do
+        cause = ->{ subject.perform }
```

Comment:
Does this read easier to you Cab? Could we use `expect { subject.perform }.to raise_error(...)` and make it a magic one liner without the use of a staby lambda ?